src/tabnineChatWidget/handlers/getEditorContextHandler.ts

```diff
@@ -58,6 +67,58 @@ export async function getEditorContext(): Promise<EditorContextResponse> {
   };
 }
 
+async function getSelectedCode(
+  editor: vscode.TextEditor,
+  request?: EditorContextRequest
+): Promise<string | undefined> {
+  const selectedCode = editor.document.getText(editor.selection);
+  if ((selectedCode && selectedCode !== "") || !request?.userQuery) {
+    return selectedCode;
+  }
+
+  const wordsInQuery = request.userQuery.split(/\s+/gm).slice(0, 20);
```

Comment:
maybe something like 
``` 
const wordRegex = /\b\w+\b/g;
const matches = text.match(wordRegex);
```
would be more percise?

Reply:
which cases are you optimising with this regex? can you give an example of a prompt that requires this regex?